### PR TITLE
[Collections] Add listPackages API

### PR DESCRIPTION
**Pull request by yim-lee**
Requires https://github.com/apple/swift-package-manager/pull/3565

Only https://github.com/apple/swift-package-manager/commit/fb8d648036e27277e5c7b0da22d367e7ec1dd8c1 is for this PR.

rdar://79528550

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/API.swift [hunk 1]
```diff
@@ -132,6 +132,16 @@ public protocol PackageCollectionsProtocol {
         callback: @escaping (Result<PackageCollectionsModel.PackageMetadata, Error>) -> Void
     )
 
+    /// Lists packages from the specified collections.
+    ///
+    /// - Parameters:
+    ///   - collections: Identifiers of the collections
+    ///   - callback: The closure to invoke when result becomes available
+    func listPackages(
+        collections: Set<PackageCollectionsModel.CollectionIdentifier>,
```
Comment (tomerd) [inline Sources/PackageCollections/API.swift:141]: should collections be optional, and when nil list from all collections?

**Review by tomerd — APPROVED** [review]

**Comment by tomerd** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/API.swift (hunk 1 above)
Comment (yim-lee) [reply Sources/PackageCollections/API.swift:141]: Yep I will make it optional

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test macOS

**Pull request merged**